core/src/main/java/org/elasticsearch/index/mapper/ParsedDocument.java

```diff
@@ -105,8 +108,13 @@ public BytesReference source() {
         return this.source;
     }
 
-    public void setSource(BytesReference source) {
+    public XContentType getXContentType() {
```

Comment:
I think this allows to remove one more deprecation warning from IndexingSlowlog. Maybe also from TermVectorService. Ideally whenever the source is retrieved, the content type should also be retrieved through its getter. Leave the Engine out as we already have an issue opened for that.